rollup/sync_service.go

```diff
@@ -1,641 +1,249 @@
-		address:                          address,
-		gasLimit:                         cfg.GasLimit,
-		txpool:                           txpool,
-		bc:                               bc,
-		eth1ChainId:                      cfg.Eth1ChainId,
-		eth1NetworkId:                    cfg.Eth1NetworkId,
-		ctcDeployHeight:                  cfg.CanonicalTransactionChainDeployHeight,
-		db:                               db,
-		clearTransactionsAfter:           (5760 * 18), // 18 days worth of blocks
-		clearTransactionsTicker:          time.NewTicker(time.Hour),
-		sequencerIngestTicker:            time.NewTicker(15 * time.Second),
-		txCache:                          NewTransactionCache(),
-		HeaderCache:                      [headerCacheSize]*types.Header{},
-	}
-	return &service, nil
-}
-
-// Start initializes the service, connecting to Ethereum1 and starting the
-// subservices required for the operation of the SyncService.
-// txs through syncservice go to mempool.locals
-// txs through rpc go to mempool.remote
-func (s *SyncService) Start() error {
-	if !s.enable {
-		return nil
-	}
-	log.Info("Initializing Sync Service", "endpoint", s.eth1HTTPEndpoint, "eth1-chainid", s.eth1ChainId, "eth1-networkid", s.eth1NetworkId, "address-resolver", s.AddressResolverAddress, "tx-ingestion-address", s.address, "confirmation-depth", s.confirmationDepth)
-	log.Info("Watching topics", "transaction-enqueued", hexutil.Encode(transactionEnqueuedEventSignature), "queue-batch-appened", hexutil.Encode(queueBatchAppendedEventSignature), "sequencer-batch-appended", hexutil.Encode(sequencerBatchAppendedEventSignature))
+		ctx:                       ctx,
+		cancel:                    cancel,
+		verifier:                  cfg.IsVerifier,
+		enable:                    cfg.Eth1SyncServiceEnable,
+		confirmationDepth:         cfg.Eth1ConfirmationDepth,
+		bc:                        bc,
+		txpool:                    txpool,
+		eth1ChainId:               cfg.Eth1ChainId,
+		client:                    client,
+		db:                        db,
+		pollInterval:              time.Second * 15,
+		timestampRefreshThreshold: time.Second * 10,
+	}
+
+	// Initial sync service setup if it is enabled. This code depends on
+	// a remote server that indexes the layer one contracts. Place this
+	// code behind this if statement so that this can run without the
+	// requirement of the remote server being up.
+	if service.enable {
+		// Ensure that the rollup client can connect to a remote server
+		// before starting.
+		err := service.ensureClient()
+		if err != nil {
+			return nil, fmt.Errorf("Rollup client unable to connect: %w", err)
+		}
 
-	// Always initialize syncing to true to start, the sequencer can toggle off
-	// syncing while the verifier is always syncing
-	s.setSyncStatus(true)
+		// Initialization logic
+		block := service.bc.CurrentBlock()
+		if block == nil {
+			return nil, errors.New("Current block is nil")
+		}
+		// TODO: temporary disable this logic on startup
```

Comment:
Turn this into a config option that defaults to false. I think the bc itself rolls back under the hood, need to test this logic